### PR TITLE
Fix BatchNorm sample handle leak issue

### PR DESCRIPTION
**Pull request by kangbintNV**
## Affected area

Documentation or samples

## Summary

Fix BatchNorm sample cuDNN handle leak issue.

## Why

Ensure the BatchNorm forward and backward paths release cuDNN handles correctly.

## Testing

Passed on B100 with cuDNN 9.25.0.8:

```bash
cmake --build build --target legacy_samples -j
valgrind --leak-check=full --show-leak-kinds=definite \
  ./build/bin/legacy_samples "Batch normalization"

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Refactor**
  * Improved cuDNN handle lifecycle management in batch normalization sample workflows.
  * Added automatic cleanup and clearer ownership handling for forward and backward operations.
  * No user-facing functionality or public interfaces were changed.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

### CODE REVIEW

**Comment by Anerudhan** [comment]
@cudnn-ci-bot run

**Review by Anerudhan — APPROVED** [review]
